pulseapi/profiles/models/categories.py

```diff
@@ -59,3 +63,85 @@ class ProgramYear(models.Model):
 
     def __str__(self):
         return self.value
+
+
+class CohortRecord(models.Model):
+    profile = models.ForeignKey(
+        'profiles.UserProfile',
+        on_delete=models.CASCADE,
+        related_name='cohort_records',
+    )
+
+    program = models.ForeignKey(
+        'profiles.ProgramType',
+        on_delete=models.PROTECT,
+        related_name='profile_cohort_records',
+    )
+
+    year = models.PositiveSmallIntegerField(
+        # TODO: Change to MaxValueValidator with callable when
+        #       we update to Django > v2.2
+        validators=[YearValidator(max_offset=2)],
+        null=True,
+        blank=True,
+    )
+
+    cohort_name = models.CharField(
+        null=True,
+        blank=True,
+        max_length=200,
+    )
+
+    def __str__(self):
+        return f'{self.profile.name} - {self.program} {str(self.year)} {self.cohort_name}'
+
+    def clean(self):
+        super().clean()
+
+        # Don't allow both the cohort and year to be empty
+        if self.year is None and not self.cohort_name:
+            raise ValidationError(
+                _('Either the year or cohort must have a value')
+            )
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+    class Meta:
+        verbose_name = 'cohort record'
+        # This meta option creates an _order column in the table
+        # See https://docs.djangoproject.com/en/1.11/ref/models/options/#order-with-respect-to for more details
+        order_with_respect_to = 'profile'
+        indexes = [
+            models.Index(fields=['profile', '_order'], name='uk_membership_profile_order'),
```

Comment:
what is `uk_`? (the only thing I can think of is the UK, which would be very strange =)

Reply:
it's a db convention (so you'll only see this in the db) that stands for "unique key". We follow this for other indexes in pulse-api as well.

Reply:
I am having a hard time finding that mentioned anywhere in the django documentation - there isn't any mention of `uk_` anywhere on the link in line 114, so can I ask where this convention is codified, and can we add a comment with that link in one of the places `uk_` is used?

Reply:
after discussion on slack, R+ if there's a comment that explains what `uk_` means